cmd/register/files.go

```diff
@@ -73,27 +87,54 @@ Override Output location prefix during registration.
 ::
 
  bin/flytectl register file  _pb_output/* -d development  -p flytesnacks --continueOnError -v v2 -l "s3://dummy/prefix"
-
+	
 Usage
 `
+	sourceCodeExtension = ".tar.gz"
 )
 
 func registerFromFilesFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
 	return Register(ctx, args, cmdCtx)
 }
 
 func Register(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
-	dataRefs, tmpDir, _err := getSortedFileList(ctx, args)
-	if _err != nil {
-		logger.Errorf(ctx, "error while un-archiving files in tmp dir due to %v", _err)
-		return _err
+	var _err error
+	var dataRefs []string
+
+	// getSerializeOutputFiles will return you all proto and  source code compress file in sorted order
+	dataRefs, tmpDir, err := getSerializeOutputFiles(ctx, args)
+	if err != nil {
+		logger.Errorf(ctx, "error while un-archiving files in tmp dir due to %v", err)
+		return err
+	}
+	logger.Infof(ctx, "Parsing file... Total(%v)", len(dataRefs))
+
+	// Validate Input files
+	sourceCode, validProto, InvalidFiles := segregateSourceAndProtos(dataRefs)
+
+	if len(InvalidFiles) > 0 {
+		return fmt.Errorf("input package have some invalid files. try to run pyflyte package again %v", InvalidFiles)
 	}
-	logger.Infof(ctx, "Parsing files... Total(%v)", len(dataRefs))
-	fastFail := !rconfig.DefaultFilesConfig.ContinueOnError
+
+	var sourceCodeName string
+	if len(sourceCode) > 0 {
+		logger.Infof(ctx, "Fast Registration detected")
+		f := strings.Split(sourceCode, "/")
+		sourceCodeName = f[len(f)-1]
+
+		rconfig.DefaultFilesConfig.AdditionalDistributionPath = getAdditionalDistributionPath(rconfig.DefaultFilesConfig.AdditionalDistributionPath, storage.GetConfig())
+		if err = uploadFastRegisterArtifact(ctx, sourceCode, sourceCodeName, rconfig.DefaultFilesConfig.AdditionalDistributionPath, rconfig.DefaultFilesConfig.Version); err != nil {
+			return fmt.Errorf("please check your Storage Config. It failed while uploading the source code. %v", err)
+		}
+		logger.Infof(ctx, "Source code successfully uploaded %v/%v ", rconfig.DefaultFilesConfig.AdditionalDistributionPath, sourceCodeName)
+	}
```

Comment:
Would have prefered small function but can be done later. Want to keep the register function small with the just outlined steps